### PR TITLE
Drool scenarios testing for case mapping.

### PR DESCRIPTION
**Pull request by isaacola-hub**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-1133

### Change description ###

Added tests for drool scenarios on case mapping in ORM.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // Combine all the user profiles into a single collection for the rules engine.
+        allProfiles = new HashSet<>();
+        usersAccessProfiles.forEach((k, v) -> allProfiles.addAll(v));
+
+
+        // Set up the rule engine for validation.
+        KieServices ks = KieServices.Factory.get();
+        KieContainer kieContainer = ks.getKieClasspathContainer();
+        this.kieSession = kieContainer.newStatelessKieSession("org-role-mapping-validation-session");
+        this.kieSession.setGlobal("DATA_SERVICE", retrieveDataService);
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java:49]: We are not using retrieveDataService to fetch any Case in ORM . Yo can remove it.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+public abstract class DroolBase {
+
+    StatelessKieSession kieSession;
+    Map<String, Set<UserAccessProfile>> usersAccessProfiles;
+    List<Command<?>> commands;
+    ExecutionResults results;
+    Set<UserAccessProfile> allProfiles;
+
+    @Mock
+    private RetrieveDataService retrieveDataService = mock(RetrieveDataService.class);
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java:33]: no use of  retrieveDataService. Please remove it.

Comment (isaacola-hub) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java]: Removed.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java (hunk 1 above)
Comment (isaacola-hub) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java]: Removed.

**Review by dharmendra8186 — APPROVED** [review]

**Review by ConorH994 — APPROVED** [review]

**Review by aashishaggarwal123 — APPROVED** [review]

**Pull request merged**